rust-doc-style-guide.md

```diff
@@ -0,0 +1,320 @@
+
+- Do NOT use italics with Espressif product names, such as ESP32.
+
+
+## Mode of Narration
+
+- Use _the first person_ (we) when introducing a tutorial or explaining how things will be done. The reader will feel like being on the same team with the authors working side by side.
+
+  > We'll start writing a program for the LM3S6965, a Cortex-M3 microcontroller. We have chosen this as our initial target because it can be emulated using QEMU so you don't need to fiddle with hardware in this section, and we can focus on the tooling and the development process.
+
+- Use _the second person_ (you) when describing what the reader should do while installing software, following a tutorial or a procedure. However, in most cases you can use imperative mood as if giving orders to the readers. It makes instructions much shorter and clearer.
+
+  > 1\. Install `cargo-generate`
+  >
+  > `cargo install cargo-generate`
+  >
+  > 2\. Generate a new project
+  >
+  > `cargo generate --git https://github.com/rust-embedded/cortex-m-quickstart`
+
+- Use _the third person_ (the user, it) when describing how things work from the perspective of hardware or software
+
+  > A driver has to be initialized with an instance of type that implements a certain `trait` of the embedded-hal which is ensured via trait bound and provides its own type instance with a custom set of methods allowing to interact with the driven device.
+
+
+## Terminology
+
+This chapter lists the terms that have inconsistencies in spelling, usage, etc.
+
+If you spot other issues with terminology, please add the terms here in alphabetical order using the formatting as follows:
+
+- _Recommended term_
+  - Avoid: Add typical phrases in which this term is found
+  - Use: Add recommended phrases
+  - Note: Add more information if needed
+
+### Recommended Terms
+
+- _Cargo_
+  - Note: always use uppercase _C_
+- ESP-IDF or esp-idf
+  - Use: esp-idf when writing about the esp-idf repo
+  - Use: ESP-IDF when writing about the ESP-IDF framework or ESP-IDF programming guide
+- `no_std`
+  - Note: see `std`
+- _Product_
+  - Avoid: Espressif SoCs
+  - Use: Espressif products
+  - Note: In the content of running applications, suggested as an umbrella term for Espressif chips, modules, development boards, etc. Otherwise, it might be potentially ambiguous that running an app on an SoC is the same as running it on a module.
+- _SoC_
+  - Avoid: _chip_
+  - Note: see also _product_
+- `std` / `no_std`
+  - Use only if necessary: bare metal development, hosted environment, Standard library, Core library
+  - Use `std` / `no_std` by default in patterns, such as:
+    - write, build, or choose between  `std` / `no_std` applications
+    - requires `std` for ...
+    - use `std` for ...
+    - compile in an `std` environment
+    - support for `no_std` requires ... (`std` support)
+    - using the `std` / `no_std` approach
+    - `std` projects / templates / features
+    - `std` / `no_std` development
```

Comment:
We've decided to proceed with `std`/`no_std` but using bare metal for `no_std` can also be used, so I think this paragraph is pretty accurate